domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2801,31 +2802,92 @@ public void selectSingle(AsyncResult<SQLConnection> conn, String sql, Handler<As
    * @param replyHandler  The query result or the failure.
    */
   public void selectSingle(String sql, JsonArray params, Handler<AsyncResult<JsonArray>> replyHandler) {
-    client.getConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
+    getSQLConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
+  }
+
+  private static String legacySql(String sql, JsonArray params) {
+    if (params.isEmpty()) {
+      return sql;
+    }
+    StringBuilder preparedSql = new StringBuilder();
+    int pos = 1;
+    for (int i = 0; i < sql.length(); i++) {
+      if (sql.charAt(i) =='?') {
+        preparedSql.append("$" + pos);
+        pos++;
+      } else {
+        preparedSql.append(sql.charAt(i));
+      }
+    }
+    return preparedSql.toString();
+  }
+
+  private static Tuple legacyArguments(JsonArray params) {
+    Tuple tuple = Tuple.tuple();
+    for (int i = 0; i < params.size(); i++) {
+      tuple.addUUID(UUID.fromString(params.getString(i)));
```

Comment:
This assumes that all parameters are UUIDs. This is the case for the current unit tests, but we cannot assume that.
Example:
```
DELETE FROM tenant_raml_module_builder.foo WHERE jsonb->>'key' = ?
```

Reply:
I don't see any possibility to support the legacy `?` syntax.
I suggest to make a breaking change and switch to `$1` syntax where people must pass in Tuple and where the Tuple elements must be a UUID, not a String, where needed.

Reply:
Yep. Can't assume it. It happened to work fine for mod-inventory-storage and unit tests. I'll change it. This will make it more work to upgrade, obviously.